packages/common/container-definitions/src/audience.ts

```diff
@@ -67,4 +73,17 @@ export interface IAudience extends EventEmitter {
 	 * does not technically have a clientId tied to an active connection to the service.
 	 */
 	getMember(clientId: string): IClient | undefined;
+
+	/**
+	 * Returns this client's clientId, if it exists. undefined if this client never connected to ordering service.
+	 * Whenever this property changes, "selfChanged" event is fired on this object.
+	 * Wheneever clients loses connection and reconnects, it will raise "connected" event at various API surfaces.
+	 * It's guranteed that such events and change of self clientId happens at the same time (syncronously, one after another).
+	 *
+	 * That said, at the moment this is experimental API. It depends on some experimental settings that might change in the future.
+	 * While it's marked experimental, you break in above described promise, where you could experience "self" being changed
+	 * (and "selfChanged" event fired) while you can't find such clientId in the list of audience clients.
+	 * @experimental
+	 */
+	readonly self: string | undefined;
```

Comment:
I think the `self` naming is too vague. Why not call this something closer to `clientId` since that's what it represents?

Reply:
Agree, but I think `clientId` is also misleading, at least at this layer. I.e. consider `getMember(clientId: string)` - here clientId is _any_ clientId, where is here it's very specific clientId. I'm not sure this difference would be obvious.
Maybe
- currentClientId
- containerClientId
?

Reply:
I like `currentClientId`. Just something that indicates what the property is without having to read the docstring comments

Reply:
Did renamed. Please take a look.